pkgs/development/libraries/optparse-bash/default.nix

```diff
@@ -0,0 +1,72 @@
+{ stdenvNoCC
+, lib
+, fetchFromGitHub
+, bash
+, gnused ? null, librarySed ? gnused
+, gawk ? null, libraryAwk ? gawk
+, coreutils ? null, libraryPrintf ? coreutils
+}:
+
+assert librarySed != null;
+assert libraryPrintf != null;
+assert libraryAwk != null;
+
+stdenvNoCC.mkDerivation {
+  pname = "optparse-bash-unstable";
+  version = "2017-06-11";
+
+  src = fetchFromGitHub {
+    owner = "nk412";
+    repo = "optparse";
+    rev = "0dfc5a3344feafc818b1fb051e4bebd195d18893";
+    sha256 = "sha256-fFDKU4ZQxrDokxwjHpXQaL3XGwdrSPugQSJbXMi6tJY=";
+  };
+
+  propagatedBuildInputs = [
+    librarySed
+    libraryAwk
+    libraryPrintf
+  ];
+
+  preBuild = ''
+    substituteInPlace optparse.bash \
+      --replace sed "${librarySed}/bin/sed" \
+      --replace awk "${libraryAwk}/bin/awk" \
+      --replace printf "${libraryPrintf}/bin/printf"
+'';
+
+  buildPhase = ''
+    runHook preBuild
```

Comment:
Why the empty buildPhase? `runHook postBuild` is also missing.

Reply:
The build phase won't start if not explicitly specified due to the lack of Makefile.

Reply:
Applied